### ISSUE TITLE
A rendering issue in newer version of libass

### ISSUE THREAD

**Issue by esterTion** [libass/libass#216]
When I was working on a subtitle of an anime, I found some weird rendering issue. I was using ffmpeg with libass 0.12.3, and vertical subtitles with karaoke effect rendering was broken, the character didn't appear in order. 
I tried recompile ffmpeg with libass 0.13.2 but the issue is still there. So I downgrade to libass 0.11.2, problem solved.
Not sure which code cause the problem, I'm not a pro about this. 
I've put the resources and ass I used in a repository [here](https://github.com/lcz970/libass_issue)

**Comment by aphirst** [comment]
I can confirm this issue too. How come you distributed the sample broken up but not as one cohesive unit? (just curious :P)

**Comment by lcz970** [comment]
Not that good in English, so I'm not sure what you are asking. :(
If you are asking about cutting a slice from source, I was using ffmpreg. First copy streams and cut a big range to prevent missing of I-frame, then reencode the part you want.

**Comment by TheOneric** [comment]
This seems to be fixed in recent git master (not fixed in release 0.14.0).

**Comment by TheOneric** [comment]
Taking a closer look at the first karaoke line (Start at `0:00:00.03` and the rendering result at `0:00:02.252` (frame 54): 
```
{\fad(0,100)\blur2}{\K37}そ{\K81}っと{\K65}開{\K39}い{\K40}た{\K20}ド{\K22}ア{\K58}の{\K25}向{\K36}こ{\K43}う{\K36}に
```

In libass 0.11, the い was roughly half-filled. *(used provided hardsub)*  
In current libass master, the た following the い is roughly half-filled.  
In libass-master with patch from #343 , the い is fully filled and た not yet filled.

What **should** happen:  
The い should be fully filled at `(37+81+65+39)*10 + 3  = 0:00:02.223` and the た should be fully filled at `(37+81+65+39+40)*10 + 3  = 0:00:02.623`. Meaning at the frame in question い should already have finished filling and た should just have started.

Only master with #343 yields the correct result here.  

![comparison](https://user-images.githubusercontent.com/1668471/84580081-f77b9e80-add3-11ea-9386-843f0aa8746c.png)
